### PR TITLE
8 macros

### LINKED ISSUE #8: Macros
> void updateMacroButton(int buttonNumber, int macroOffset, bool isPressed, bool isRunning) {

If I am reading this correctly then the button text and the button color should change when a macro button is pressed

From what I can tell they never change from TFT_DARKGREY and WHITE

### PR DESCRIPTION
**Pull request by helo-head**
In theory the new changes to both the buttonWasPressed() updatemacrobutton() should resolve the issues I noted earlier. I kept isPressed as it still seems to work. If we run into issues it can always be replaced with  true/false.

### CODE REVIEW

**Comment by brianteeman** [comment]
will test in the morning and feedback

**Comment by brianteeman** [comment]
This seems to work. As I press a macro I can see it change colors and then revert on release. So the indication that you have pressed a macro button is correct.

For my own purposes I will probably add some code to indicate the last macro button pressed as I am using the macro to set the atem on air and there is no method in the skaarhoj library to get the onair status

File: Tallly-Core2-V11.ino [hunk 1]
```diff
@@ -767,26 +767,33 @@ String ip2Str(IPAddress ip) {
   return s;
 }  // End of ip2St
 
-// Button press function
+/// Button press function
```
Comment (brianteeman) [inline Tallly-Core2-V11.ino]:
```suggestion
// Button press function
```

File: Tallly-Core2-V11.ino [hunk 2]
```diff
@@ -767,26 +767,33 @@ String ip2Str(IPAddress ip) {
...
 void buttonWasPressed(Event &e) {
   Button &tbtn = *e.button;
   const char *buttonName = e.objName();
   char buttonType = buttonName[0];
   int buttonNumber = buttonName[1] - '0';
 
-  if (!postConfig) {
+  if (!postConfig) {                  //Update file selection button appropriatly
```
Comment (brianteeman) [inline Tallly-Core2-V11.ino]:
```suggestion
  if (!postConfig) {                  // Update file selection button appropriately
```

File: Tallly-Core2-V11.ino [hunk 3]
```diff
@@ -767,26 +767,33 @@ String ip2Str(IPAddress ip) {
...
     fileNumber = buttonName[1] - 48;  // Hack to convert ascii char to long
     M5.Lcd.fillRect(tbtn.x, tbtn.y, tbtn.w, tbtn.h, tbtn.isPressed() ? WHITE : BLACK);
 
   } else {
-
+  if ( e.type == E_TOUCH) {           // If a touch event set ATEM appropriatly and update the button if a maco
```
Comment (brianteeman) [inline Tallly-Core2-V11.ino]:
```suggestion
  if ( e.type == E_TOUCH) {           // If a touch event set ATEM appropriately and update the button if a macro
```

File: Tallly-Core2-V11.ino [hunk 4]
```diff
@@ -767,26 +767,33 @@ String ip2Str(IPAddress ip) {
...
     if (buttonType == 'c') {
       AtemSwitcher.changeProgramInput(buttonNumber);
     } else {
-      updateMacroButton(buttonNumber, macroOffset, tbtn.isPressed(), false);
+      updateMacroButton(buttonNumber, macroOffset, tbtn.isPressed());
       AtemSwitcher.setMacroAction(buttonNumber + macroOffset, 0);
     }
+  } else {                            // Release handler for macros to update button appropriatly
```
Comment (brianteeman) [inline Tallly-Core2-V11.ino]:
```suggestion
  } else {                            // Release handler for macros to update button appropriately
```

**Review by brianteeman — CHANGES_REQUESTED** [review]
just typos - no code change

File: Tallly-Core2-V11.ino [hunk 5]
```diff
@@ -767,26 +767,33 @@ String ip2Str(IPAddress ip) {
...
+    if (buttonType == 'm') {
+      updateMacroButton(buttonNumber, macroOffset, tbtn.isPressed());
+    }
   }
+    
+  } //End camera/macro button handling
```
Comment (brianteeman) [inline Tallly-Core2-V11.ino]:
```suggestion
  } // End camera/macro button handling
```

**Comment by helo-head** [comment]
> For my own purposes I will probably add some code to indicate the last macro button pressed as I am using the macro to set the atem on air and there is no method in the skaarhoj library to get the onair status

Actually that makes sense, I like it. Seems like it would be easy enough to add, just a call to draw all the macrobutons on TOUCH effectively resetting the last one called and then a change in color on RELEASE to reflect it was the last one called.... I can add this in if you like ?

**Comment by brianteeman** [comment]
Yes that would be great if you could.

I also found an interesting [modification ](https://github.com/AronHetLam/ATEM_tally_light_with_ESP8266) to the bmd library that adds support for detecting the streaming status. 

That would allow for an **ON AIR** label

**Pull request merged**